packages/clerk-js/src/ui/components/ApiKeys/ApiKeysTable.tsx

```diff
@@ -0,0 +1,226 @@
+import { useEffect, useState } from 'react';
+import useSWR from 'swr';
+
+import { timeAgo } from '../../../utils/date';
+import {
+  Box,
+  Button,
+  descriptors,
+  Flex,
+  Icon,
+  Input,
+  localizationKeys,
+  Spinner,
+  Table,
+  Tbody,
+  Td,
+  Text,
+  Th,
+  Thead,
+  Tr,
+} from '../../customizables';
+import { ThreeDotsMenu } from '../../elements';
+import { useClipboard } from '../../hooks';
+import { Check, ClipboardOutline, Eye, EyeSlash } from '../../icons';
+import { common, mqu } from '../../styledSystem';
+
+const useApiKeySecret = ({ apiKeyID, enabled }: { apiKeyID: string; enabled: boolean }) => {
+  const clerk = useClerk();
+
+  return useSWR(enabled ? ['api-key-secret', apiKeyID] : null, () => clerk.getApiKeySecret(apiKeyID));
+};
+
+const CopySecretButton = ({ apiKeyID }: { apiKeyID: string }) => {
+  const [enabled, setEnabled] = useState(false);
+  const { data: apiKeySecret } = useApiKeySecret({ apiKeyID, enabled });
+  const { onCopy, hasCopied } = useClipboard(apiKeySecret ?? '');
+
+  useEffect(() => {
+    if (enabled && apiKeySecret) {
+      onCopy();
+      setEnabled(false);
+    }
+  }, [enabled, apiKeySecret, onCopy]);
+
+  return (
+    <Button
+      variant='ghost'
+      aria-label={hasCopied ? 'Copied API key to clipboard' : 'Copy API key'}
+      onClick={() => setEnabled(true)}
+      focusRing={false}
+    >
+      <Icon
+        size='sm'
+        icon={hasCopied ? Check : ClipboardOutline}
+        sx={t => ({ color: t.colors.$primary500 })}
+      />
+    </Button>
+  );
+};
+
+const SecretInputWithToggle = ({ apiKeyID }: { apiKeyID: string }) => {
+  const [revealed, setRevealed] = useState(false);
+  const { data: apiKeySecret } = useApiKeySecret({ apiKeyID, enabled: revealed });
+
+  return (
+    <Flex
+      center
+      sx={{
+        width: '100%',
+        position: 'relative',
+      }}
+    >
+      <Input
+        type={revealed ? 'text' : 'password'}
+        value={revealed && apiKeySecret ? apiKeySecret : '•••••••••••••••••••••••••'}
+        readOnly
+        aria-label='API key (hidden)'
+        sx={t => ({
+          paddingInlineEnd: t.sizes.$12,
+        })}
+      />
+      <Button
+        variant='ghost'
+        sx={t => ({
+          position: 'absolute',
+          right: 0,
+          '&:focus-visible': {
+            ...common.focusRingStyles(t),
+          },
+        })}
+        focusRing={false}
+        aria-label={'Show key'}
+        onClick={() => setRevealed(!revealed)}
+      >
+        <Icon
+          icon={revealed ? EyeSlash : Eye}
+          sx={t => ({ color: t.colors.$colorTextSecondary })}
+        />
+      </Button>
+    </Flex>
+  );
+};
+
+export const ApiKeysTable = ({
+  rows,
+  isLoading,
+  onRevoke,
+}: {
+  rows: APIKeyResource[];
+  isLoading: boolean;
+  onRevoke: (id: string, name: string) => void;
+}) => {
+  return (
+    <Flex sx={t => ({ overflowX: 'auto', padding: t.space.$1, width: '100%' })}>
```

Comment:
The padding usage here is causing some alignment issues:

![Screenshot 2025-05-27 at 9 46 59 AM](https://github.com/user-attachments/assets/4d136ea8-a345-4566-aaab-ad77cc4ec685)

Reply:
Fixed! Thanks